src/ray/raylet/worker_pool.cc

```diff
@@ -746,7 +734,7 @@ Status WorkerPool::RegisterDriver(const std::shared_ptr<WorkerInterface> &driver
   auto &state = GetStateForLanguage(driver->GetLanguage());
   state.registered_drivers.insert(std::move(driver));
   const auto job_id = driver->GetAssignedJobId();
-  all_jobs_[job_id] = job_config;
+  HandleJobStarted(job_id, job_config);
```

Comment:
I don't have much context here, do you know why HandleJobStarted wasn't called here previously?  Was this a bug?

Reply:
+1 to Archit's comment. Either way, can we get a unit test to ensure the ref counting behavior is correct and stays correct?

Reply:
`HandleJobStarted` only do two things: First, add job to `all_jobs_ `. Second, install runtime env eagerly if needed.

Before current PR:
`HandleJobStarted` will be triggered by the publishing channel of job. So here, in `RegisterDriver`, it's ok to only add job to `all_jobs_`.

In current PR:
To guarantee the create runtime env RPC only be called once, I have modified the logic of  `HandleJobStarted`. `HandleJobStarted` will check if the job already been added. So, I also use `HandleJobStarted` here.